src/convertToObject.js

```diff
@@ -12,7 +12,28 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const propertiesAndValuesArray = sourceString.split(';');
+
+  for (let i = 0; i < propertiesAndValuesArray.length; i++) {
+    propertiesAndValuesArray[i] = propertiesAndValuesArray[i].trim();
+  }
+
+  const filteredPropAndVal = propertiesAndValuesArray.filter(
+    item => item !== ''
+  );
+
+  for (let i = 0; i < filteredPropAndVal.length; i++) {
+    filteredPropAndVal[i] = filteredPropAndVal[i].split(':');
+  }
```

Comment:
let's use array iteration methods in this task instead of for loops (map, forEach)